File: leetcode/problem1678.js
/**
 * @param {string} command
 * @return {string}
 */
var interpret = function(command) {
    let str = ""
    for(let i = 0; i < command.length; i++) {
        if(command[i] === "G") {
            str += "G"
        }
        if(command[i] === "(" && command[i + 1] === ")") {
            str+= "o"
        }
        if(command[i] === "(" && command[i + 1] === "a"){
            str += "al"
        }
    }
    return str
};

// Not gonna lie...this approach seems bad. I'm going to try to work on a better solution, which will likely involve this:
    // let pieces = {
    //     "G": "G",
    //     "()": o,
    //     "(al)": al
    // }

// https://leetcode.com/problems/goal-parser-interpretation/

File: leetcode/problem1678.go
func interpret(command string) string {
    var interpretation string
    for i, _ := range command {
        switch string(command[i]) {
        case "G":
        interpretation += "G"
        case "(":
        switch string(command[i + 1]) {
            case ")":
            interpretation += "o"
            case "a":
            interpretation += "al"
        }
    }
    }
    return interpretation
}

// This solution beat 100% of other solutions in time complexity the first time I submitted it

// https://leetcode.com/problems/goal-parser-interpretation/
